bin/yb-ctl

```diff
@@ -1022,48 +1070,135 @@ class ClusterControl:
-                # aware of it because we do not have a yb-admin API to return only live tablet
-                # servers.
                 if num_yb_admin_ts == num_alive_ts:
+                    # This will not work if you have stopped/removed a node and the master is still
+                    # aware of it because we do not have a yb-admin API to return only live tablet
+                    # servers.
                     return True
+                elif num_yb_admin_ts < 0:
+                    # We might get no output from yb-admin if the leader is not up yet...
+                    logging.info("Master leader election still pending...")
+                else:
+                    logging.info("Waiting for all tablet servers to register: {}/{}".format(
+                        num_yb_admin_ts, num_alive_ts))
             except subprocess.CalledProcessError:
                 pass
-            wait_count += 1
-            time.sleep(SLEEP_TIME_IN_SEC)
-        logging.error("Failed waiting for {} tservers, got {}".format(
-            num_alive_ts, num_yb_admin_ts))
-        return False
+
+        logging.info("Waiting for master leader election tablet server registration.")
+        if not retry_fn_with_timeout(call_check_for_ts, self.options.timeout):
+            logging.error("Failed waiting for {} tservers, got {}".format(
+                num_alive_ts, num_yb_admin_ts))
+            return False
+        return True
 
     def wait_for_cluster_or_raise(self):
         if not self.wait_for_cluster():
             raise RuntimeError("Timed out waiting for Yugabyte cluster!")
 
-    def show_node_status(self, daemon_id):
-        pid = self.get_pid(daemon_id)
-
-        def get_address(port_type):
-            return self.options.get_address(daemon_id, port_type, include_base_url=True)
-
-        if pid is None:
-            logging.info("Server {} is not running".format(daemon_id))
-        else:
-            info_list = [
-                ("type", daemon_id.daemon_type),
-                ("node_id", daemon_id.index),
-                ("PID", pid),
-                ("admin service", "http://" + get_address("http"))
-            ]
-
-            if daemon_id.is_tserver():
-                info_list.extend([
-                    ("cql service", get_address("cql_rpc")),
-                    ("redis service", get_address("redis_rpc")),
-                ])
-                if self.is_postgres_enabled():
-                    info_list.append(("pgsql service", get_address("pgsql_rpc")))
-            logging.info("Server is running: {}".format(
-                ", ".join(["{}={}".format(k, v) for k, v in info_list])))
+    def query_cluster_uuid(self):
+        cluster_uuid = None
+        try:
+            with open(os.devnull, "wb") as devnull:
+                cluster_uuid = subprocess.check_output(
+                    "curl localhost:7000/cluster-config | grep cluster_uuid",
```

Comment:
Let's not add a dependency on curl -- let's send the HTTP call in Python.

Reply:
I just removed this all together, since we decided not to do the `cluster_uuid` at all